src/main/java/uk/gov/companieshouse/api/accounts/interceptor/SmallFullInterceptor.java

```diff
@@ -0,0 +1,125 @@
+package uk.gov.companieshouse.api.accounts.interceptor;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerMapping;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+import uk.gov.companieshouse.api.accounts.AttributeName;
+import uk.gov.companieshouse.api.accounts.CompanyAccountsApplication;
+import uk.gov.companieshouse.api.accounts.LinkType;
+import uk.gov.companieshouse.api.accounts.model.entity.CompanyAccountEntity;
+import uk.gov.companieshouse.api.accounts.model.entity.SmallFullEntity;
+import uk.gov.companieshouse.api.accounts.service.SmallFullService;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+/**
+ * The SmallFullInterceptor interceptor validates that the Smallfull entity is in the correct state
+ * for the calling request. It insures a Smallfull resource exists and matches the one stored in the
+ * CompanyAccount provided in the session. This interceptor will fail if the CompanyAccount
+ * interceptor has not run prior to its own execution.
+ */
+@Component
+public class SmallFullInterceptor extends HandlerInterceptorAdapter {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(CompanyAccountsApplication.APPLICATION_NAME_SPACE);
+
+    @Autowired
+    private SmallFullService smallFullService;
+
+    /**
+     * This class validates a Small-full account exists for the given CompanyAccount. Validation is
+     * carried out via a database lookup for the SmallFullEntity Object then matching the retrieved
+     * Entities 'self' link to the session stored CompanyAccountEntities 'small_full_accounts' link.
+     * Providing this validation passes it assigns the SmallFullEntity to the session.
+     *
+     * @param request - current HTTP request
+     * @param response - current HTTP response
+     * @param handler - chosen handler to execute, for type and/or instance evaluation
+     * @return true if the execution chain should proceed with the next interceptor or the handler
+     * itself.
+     */
+    @Override
+    @SuppressWarnings("unchecked")
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response,
+            Object handler) throws NoSuchAlgorithmException {
+
+        if (request.getMethod().equalsIgnoreCase("POST") && request.getRequestURI()
```

Comment:
probably worth adding a comment on this case to explain this it's to avoid trying to get a small-full resource when the request is trying to create one

Reply:
Done.